src/plugins/intel_npu/src/plugin/CMakeLists.txt

```diff
@@ -56,11 +57,21 @@ target_link_libraries(${TARGET_NAME}
 target_include_directories(${TARGET_NAME}
     PRIVATE
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    PRIVATE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/>
 )
 target_include_directories(${TARGET_NAME}
     SYSTEM PRIVATE
         ${NPU_PLUGIN_SOURCE_DIR}/thirdparty/level-zero/include
         ${NPU_PLUGIN_SOURCE_DIR}/thirdparty/level-zero-ext
 )
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  target_compile_options(${TARGET_NAME} PRIVATE -mavx2 -mf16c)
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+  target_compile_options(${TARGET_NAME} PRIVATE /arch:AVX2)
+else()
+  message(AUTHOR_WARNING "Unknown compiler, may miss the AVX2 baseline setting")
+endif()
```

Comment:
could you please use this function instead?
https://github.com/openvinotoolkit/openvino/blob/3d08b4907dd2175f9b71c6714e866bf46b8b4c72/cmake/developer_package/compile_flags/os_flags.cmake#L124-L138

Reply:
Just noticed, will add a task to cover it